frontend/lib/pages/cross-site-terms-opt-in.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useContext } from "react";
+import {
+  MiddleProgressStep,
+  ProgressStepRoute,
+} from "../progress/progress-step-route";
+import Page, { SiteName } from "../ui/page";
+import { PrivacyPolicyLink, TermsOfUseLink } from "../ui/privacy-info-modal";
+import { SessionUpdatingFormSubmitter } from "../forms/session-updating-form-submitter";
+import { AgreeToTermsMutation } from "../queries/AgreeToTermsMutation";
+import { AppContext, getGlobalAppServerInfo } from "../app-context";
+import { ProgressButtons } from "../ui/buttons";
+import { CheckboxFormField, HiddenFormField } from "../forms/form-fields";
+import { AllSessionInfo } from "../queries/AllSessionInfo";
+import { isUserLoggedIn } from "../util/session-predicates";
+
+export function hasLoggedInUserAgreedToTerms(s: AllSessionInfo): boolean {
+  if (!s.onboardingInfo) {
+    console.warn("Logged-in user has no onboarding info!");
+    return false;
+  }
+  const { siteType } = getGlobalAppServerInfo();
+  switch (siteType) {
+    case "JUSTFIX":
+      return s.onboardingInfo.agreedToJustfixTerms;
+
+    case "NORENT":
+      return s.onboardingInfo.agreedToNorentTerms;
+  }
+}
+
+export const CrossSiteAgreeToTerms = MiddleProgressStep((props) => {
+  const { siteType } = useContext(AppContext).server;
+
+  return (
+    <Page title="Please agree to our privacy policy" withHeading="big">
```

Comment:
I would actually change the title to "Please agree to our terms of use" since we're not only focused on privacy policy here. What do you think?

Reply:
Oh good catch! I initially started this PR thinking it was just the privacy policy but realized it was t & c halfway through... renamed all the variables but forgot about this title!